### PR TITLE
AI code policy, and AI-assissted function rewrites

### PR DESCRIPTION
**Pull request by jkirk5**
### Summary

Adds documentation page on our AI use policy.

Used AI (through cursor, which does not disclose models used) to resolve 2 TODOs in engine_deck.py by modifying existing code. `extend_array()` and `_hybrid_throttle_norm()` are the two functions with AI-written code.

Some additional manual cleanup, review, and formatting of that file, including updated docstrings to reflect new code.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Review by hschilling — APPROVED** [review]
We can always change it but it is important to get our best effort policy out there sooner rather than later.

File: aviary/docs/developer_guide/ai_policy.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Any use of AI tools should be clearly disclosed. It is helpful to separate AI-generated code into their own commits, but is not always possible. Mention that AI was used in the commit messages. Give additional details in your pull request descriptions about how AI was used. Include a brief explanation of what you used the AI for (debugging, generating new code, etc.), which model you used, and which files were affected.
+
+## 2. Thoroughly test all AI-generated code
+
+Like all code contributions, it is important to run all unittests and benchmark tests on as many different OS and environment configurations as you reasonably can - this goes double for any changes that used AI assistance. If any code generated by AI is not explicitly tested by a unittest, you should write a new test to cover the AI generated code. If AI modifies existing code, especially if new functionality is added, existing unittests should be extended to test that new functionality. 
+
+## 3. AI should never be used as a source for data
+
+
```
Comment (Kenneth-T-Moore) [inline aviary/docs/developer_guide/ai_policy.md:16]: extra line

File: aviary/docs/developer_guide/ai_policy.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Aviary's AI Policy
+Use of generative artificial intelligence (LLMs) to produce code for Aviary is allowed, but contributions must follow the following guidelines. If you wish to use AI to assist in contributing code to Aviary, we recommend sticking to limited, targeted uses to speed up development time for small-scale tasks, or to limit AI use to a purely advisory role and not for writing code. Extensive use of AI-generated code leads to large, fragmented, and difficult to maintain codebases.
+
+However, this advice is based on experience using current-generation AI models, and as these tool's capabilities improve we may revise our recommendations.
+
+## 1. Disclose any use of AI
+
+Any use of AI tools should be clearly disclosed. It is helpful to separate AI-generated code into their own commits, but is not always possible. Mention that AI was used in the commit messages. Give additional details in your pull request descriptions about how AI was used. Include a brief explanation of what you used the AI for (debugging, generating new code, etc.), which model you used, and which files were affected.
```
[the remark this answers is not in the document]
Comment (jkirk5) [reply aviary/docs/developer_guide/ai_policy.md:8]: Cursor doesn't disclose which model(s?) it uses, so I don't actually have that information

File: aviary/docs/developer_guide/ai_policy.md (hunk 1 above)
Comment (jkirk5) [inline aviary/docs/developer_guide/ai_policy.md:16]:
```suggestion
```

File: aviary/docs/developer_guide/ai_policy.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (jkirk5) [inline aviary/docs/developer_guide/ai_policy.md:8]:
```suggestion
Any use of AI tools should be clearly disclosed. It is helpful to separate AI-generated code into their own commits whenever possible. Mention that AI was used in the commit messages. Give additional details in your pull request descriptions about how AI was used. Include a brief explanation of what you used the AI for (debugging, generating new code, etc.), which model you used, and which files were affected.
```

File: aviary/docs/developer_guide/ai_policy.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+## 2. Thoroughly test all AI-generated code
+
+Like all code contributions, it is important to run all unittests and benchmark tests on as many different OS and environment configurations as you reasonably can - this goes double for any changes that used AI assistance. If any code generated by AI is not explicitly tested by a unittest, you should write a new test to cover the AI generated code. If AI modifies existing code, especially if new functionality is added, existing unittests should be extended to test that new functionality. 
+
+## 3. AI should never be used as a source for data
+
+Similarly, AI should not be used to write unittests. While writing unittests can be a tedious step that may be tempting to automate, they are extremely critical to get right. The test setup and which values are used must be carefully chosen to ensure the code is properly being verified (and potentially validated) including potential edge cases.
+
+## 3. Do not use AI for documentation
```
Comment (jkirk5) [inline aviary/docs/developer_guide/ai_policy.md:18]:
```suggestion
## 4. Do not use AI for documentation
```

File: aviary/docs/developer_guide/ai_policy.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+AI tools should not be used for documentation. Current AI tools struggle with holding large codebases like Aviary in context, and hallucinations or general mistakes are harder to catch in text then they are in code - even with Aviary's documentation checking utilities (DocTAPE). Any AI written documentation pages will be rejected.
+
+## 4. Use common sense when using AI tools
```
Comment (jkirk5) [inline aviary/docs/developer_guide/ai_policy.md:22]:
```suggestion
## 5. Use common sense when using AI tools
```

**Review by ehariton — APPROVED** [review]
If we want to disclose use of AI, we should add a checkbox and and prompt on the PR page. 

### Was AI Used In this PR Y/N

- Was AI Used to Generate Data Y/N
- Did you use AI to generate unit tests Y/N
- What unit tests cover this code?
- Did you use AI to generate documentation Y/N

**Comment by xjjiang** [comment]
Given a component, can AI help summarize its functionality? Can AI create necessary unit tests or at least give a list of necessary unit tests?

**Comment by jkirk5** [comment]
> Given a component, can AI help summarize its functionality? Can AI create necessary unit tests or at least give a list of necessary unit tests?

AI is decent at summarizing code, so it may be useful as a starting place for writing docstrings. AI should not be used for writing unittests (policy item 3), since we need to be careful about which cases we run and have good values for our "truth" data
